cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -100,39 +100,37 @@ struct EnumTraits<compute::CompareOperator>
   }
 };
 template <>
-struct EnumTraits<compute::AssumeTimezoneOptions::Ambiguous>
-    : BasicEnumTraits<compute::AssumeTimezoneOptions::Ambiguous,
-                      compute::AssumeTimezoneOptions::Ambiguous::AMBIGUOUS_RAISE,
-                      compute::AssumeTimezoneOptions::Ambiguous::AMBIGUOUS_EARLIEST,
-                      compute::AssumeTimezoneOptions::Ambiguous::AMBIGUOUS_LATEST> {
+struct EnumTraits<compute::Ambiguous>
+    : BasicEnumTraits<compute::Ambiguous, compute::Ambiguous::AMBIGUOUS_RAISE,
+                      compute::Ambiguous::AMBIGUOUS_EARLIEST,
+                      compute::Ambiguous::AMBIGUOUS_LATEST> {
```

Comment:
I'm not sure if `AssumeTimezoneOptions::Ambiguous` and `RoundTemporalOptions::Ambiguous` would have the same options long-term (same for nonexistent). For now this change seems like the way to go, I'm just wondering if the name `compute::Ambiguous` should maybe be `compute::AmbiguousTime` (and `compute::NonexistentTime`?